engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h

```diff
@@ -169,6 +189,21 @@ class SortAndSampleKernel : public kernel {
 				sampledTables.push_back(std::move(sampledTable));
 				localTotalNumRows += batch->view().num_rows();
 				localTotalBytes += batch->sizeInBytes();
+
+				// Try samples estimation
+				if(try_num_rows_estimation) {
+					uint64_t num_rows_estimate;
+					std::tie(estimate_samples, num_rows_estimate) = this->query_graph->get_estimated_input_rows_to_cache(this->get_id(), std::to_string(this->get_id()));
+					num_sample_rows_estimate = static_cast<uint64_t>(num_sample_rows_estimate * ORDER_BY_SAMPLES_RATIO);
+					try_num_rows_estimation = false;
+				}
+				sum_sample_rows += sampledTables.back()->num_rows();
```

Comment:
We are not trying to collect how many samples, but how big is the population we sampled. So this should be:
population_sampled += batch->num_rows();